### ISSUE TITLE
Processing: Not completely covered by `py.test -k Processing`

### ISSUE THREAD

**Issue by Makman2** [coala/coala#3345]
```
mak@hp-elitebook-2540p:~/dev/coala (Makman2/quotes-bear-fail)$ py.test -k Processing --cov | grep Processing.py
coalib/processes/Processing.py                                        226      2     94      0    99%   684-685
```
Tests should be completely self-covering.

**Comment by Shade5** [comment]
I'm a newcomer, will I be able to fix this? If so please assign me to this.

**Comment by Makman2** [comment]
sure you can try :)

You need to write tests / modify existing tests in `ProcessingTest.py` so that every piece of code is covered when running `py.test -k Processing`.

**Comment by Shade5** [comment]
I installed the  package **python-pytest**
However, I'm getting the following error when I run your command
`py.test -k Processing --cov`


```
usage: py.test [options] [file_or_dir] [file_or_dir] [...]
py.test: error: unrecognized arguments: --cov
  inifile: None
  rootdir: /home/a/workspace/coala/coala
```

**Comment by Mixih** [comment]
As discussed on gitter(and here just for future reference), `# pragma no cover` is almost never the way to raise coverage as it doesn't actually test the code. You always want to write tests where possible

**Comment by Shade5** [comment]
@Mixih Yes, got it. I just forgot to update the comment

**Comment by Shade5** [comment]
The part of the code that needs to be covered is in a try catch statement. I was trying to use assertRaises as suggested on gitter. But the exception was never caught as the it gets caught in the try catch itself. So I think I just have to trigger the exception for full coverage. 

Please let me know if I'm wrong

**Comment by Makman2** [comment]
sounds good :+1: @Shade5 :)

**Comment by Mixih** [comment]
yep, to test exceptions you need to trigger the correct exception you want in the code under test(uut). Wrap the code that should trigger the exception with the assertRaisesRegex statement(we don't use assertRaises anymore because assertRaisesRegex verifies it's the correct exception)

**Comment by Shade5** [comment]
@Mixih @Makman2  Here's the pull request  
https://github.com/coala/coala/pull/3390
